zipline/assets/assets.py

```diff
@@ -338,6 +446,8 @@ def spawn_asset(self, identifier, **kwargs):
         self.identifier_cache[identifier] = asset
         if asset.symbol is not '':
             self.sym_cache.setdefault(asset.symbol, []).append(asset)
+        if isinstance(asset, Future) and asset.root_symbol is not '':
+            self.future_cache.setdefault(asset.root_symbol, []).append(asset)
```

Comment:
Though, after re-reading this function, I don't think we should be putting anything in any caches here; we should just create and return the asset, and then `populate_caches`, which is the only place that calls this, should be responsible for adding assets to the caches.